### PR TITLE
Support multiple mock OIDC configs per frontend portal

### LINKED ISSUE #284: Support multiple mock OIDC configs per frontend portal in @cellix/server-oauth2-mock-seedwork
### Summary

Enhance @cellix/server-oauth2-mock-seedwork and @apps/server-oauth2-mock so each @apps/ui-* portal can define one or more mock OIDC configs.

Today, discovery effectively supports only one OIDC config per portal. The mock OAuth2 server already supports multiple configs internally, so this task should simplify the app-level config/discovery model to expose that capability.

### Desired Behavior

Update mock-oidc.json so it always contains an array of OIDC config definitions.

- Portals with one login provide a one-element array.
- Portals with multiple logins provide multiple config entries.
- Discovery should return/register one mock OIDC config per array element.
- Explicit envVars mappings remain the source of truth for resolving OIDC values.
- User account claims should continue to come from mock-oidc.users.json by default.
- claims should not be required in mock-oidc.json.

### Naming Updates

Update the existing sample portal configs to use clearer OIDC config names:

- ui-community config name: end-user
- ui-staff config name: staff-user

Update related env vars, tests, docs, and route/discovery references accordingly.

### Env Var Convention

Configured frontend OIDC env var names should follow the standard Cellix convention:

```txt
VITE_APP_<PORTAL>_<OIDC_CONFIG_NAME>_<AAD|B2C>_<AUTHORITY|CLIENT_ID|REDIRECT_URI> 
```

Required sample naming:

- VITE_APP_UI_COMMUNITY_END_USER_B2C_*
- VITE_APP_UI_STAFF_STAFF_USER_AAD_*

The implementation should validate this convention but should not derive env var names implicitly.

### Scope

- Change mock-oidc.json parsing to expect an array.
- Update existing mock-oidc.json files to the array shape.
- Preserve explicit envVars mapping behavior.
- Preserve existing mock-oidc.users.json user/claims behavior.
- Preserve owning portal directory context for .env, .env.local, local overrides, and mock users.
- Ensure each OIDC config has a distinct safe name for issuer/discovery routing.
- Detect duplicate config names intentionally instead of silently overwriting.
- Update documentation for the array-based config shape and env var convention.

### Testing Requirements

Add or update tests to verify:

- Single-element config arrays discover/register one OIDC config.
- Multi-element config arrays discover/register multiple OIDC configs for the same portal.
- Each config resolves authority, clientId, and redirectUri through its explicit envVars mappings.
- ui-community resolves the end-user B2C env vars correctly.
- ui-staff resolves the staff-user AAD env vars correctly.
- User claims continue loading from mock-oidc.users.json.
- claims are not required in mock-oidc.json.
- Env var names are validated against the Cellix convention.
- Duplicate config names are handled with a clear validation error or warning.
- Existing @ocom mock OAuth2 behavior does not regress.

### Acceptance Criteria

- @apps/server-oauth2-mock supports one or more mock OIDC configs per portal through an array-based mock-oidc.json.
- Existing sample portals are updated to end-user and staff-user.
- Explicit envVars mappings remain the source of truth.
- Default user claims continue to come from mock-oidc.users.json.
- Discovery registers one mock OIDC config per array element.
- Tests cover single-config, multi-config, env var mapping, env var convention validation, user claims loading, duplicate names, and current @ocom behavior.
- Documentation reflects the new config shape.

### PR DESCRIPTION
**Pull request by noce-nick**
Enhance the mock OIDC configuration to support an array-based structure in `mock-oidc.json`, allowing each frontend portal to define multiple OIDC configurations. Update sample configurations for `ui-community` and `ui-staff` to reflect clearer naming conventions. Implement validation for environment variable names and ensure backward compatibility with existing configurations. Add tests to verify the functionality of single and multiple OIDC configurations, including handling of duplicate names and environment variable mapping. Update documentation to reflect the new configuration schema and naming conventions.

Fixes #284

## Summary by Sourcery

Support multiple mock OIDC configurations per UI portal and centralize reusable mock OAuth2 utilities into the shared seedwork package while updating configurations and tests to the new naming and discovery schemes.

New Features:
- Allow mock-oidc.json to define multiple OIDC configurations per UI portal via array entries, each registering as a distinct issuer.
- Expose reusable helpers from the server-oauth2-mock seedwork package, including portal discovery, filesystem-backed user store, and URL port normalization.

Bug Fixes:
- Ensure URL port injection preserves all URL components and avoids adding protocol-default ports.
- Prevent transient Mongo transaction errors in tests by pre-creating collections before transactional scenarios.
- Allow mock OIDC portal discovery to fall back to injected environment variables when .env files are missing or unreadable, instead of always skipping portals.

Enhancements:
- Refine mock OIDC portal registration names to be derived from portal directory and config name, with validation for duplicates and invalid names.
- Warn but continue processing when mock OIDC env var names do not follow project naming conventions, and skip only invalid mock-oidc.json elements.
- Improve queue storage registration by simplifying option merging and tightening TypeScript types in consumer/producer utilities.
- Move the file-backed mock user store into the shared seedwork and update references and tests accordingly.

Build:
- Update CI and local dev environment variables to the new, more explicit OIDC env var naming scheme for community and staff portals.
- Adjust the monorepo verify pipeline to omit the Sonar quality gate check from the default verification command.

Documentation:
- Extend mock OAuth2 server and seedwork documentation to describe multi-config mock-oidc.json support, new naming rules, and exported utilities.

Tests:
- Add comprehensive tests for multi-entry mock-oidc.json arrays, duplicate and invalid registration names, env resolution precedence, and non-conforming env var warnings.
- Expand coverage for ensurePortInUrl behavior, including default-port handling and IPv6 URLs.
- Update UI, API, and queue storage tests to reflect new env var names, portal issuers, and refactored helper locations.

Chores:
- Align UI, API, and E2E configuration, storybook mocks, and local-dev scripts with the new OIDC issuer paths and environment variable names.
- Tidy formatting and minor TypeScript style issues across queue storage and persistence packages.

### CODE REVIEW

File: packages/cellix/server-oauth2-mock-seedwork/tests/portal-discovery.test.ts [hunk 1]
```diff
@@ -358,4 +399,83 @@ describe('discoverPortalConfigs', () => {
...
+		});
+		writeEnv(tmp, 'ui-nonconform/.env', 'CLIENT_NC=cid-nc\nREDIR_NC=https://nc/redirect\n');
+
+		const warnSpy = vi.spyOn(console, 'warn').mockImplementation(() => undefined);
+		try {
+			const portals = discoverPortalConfigs(tmp);
+			expect(portals).toHaveLength(1);
+			expect(warnSpy).toHaveBeenCalled();
+			const p = portals[0];
+			expect(p?.clientId).toBe('cid-nc');
```
[the remark this answers is not in the document]
Comment (noce-nick) [reply packages/cellix/server-oauth2-mock-seedwork/tests/portal-discovery.test.ts:567]: Good suggestion, done

File: packages/cellix/server-oauth2-mock-seedwork/tests/portal-discovery.test.ts [hunk 2]
```diff
@@ -56,10 +56,51 @@ describe('discoverPortalConfigs', () => {
...
+	it('supports multi-config mock-oidc.json arrays and resolves envs per element', () => {
+		if (!tmp) throw new Error('tmp not created');
+
+		// Single ui dir with multiple configs
+		writeJson(tmp, 'ui-community/mock-oidc.json', [
+			{
+				name: 'end-user',
+				envVars: { clientId: 'VITE_APP_COMMUNITY_END_CLIENTID', redirectUri: 'VITE_APP_COMMUNITY_END_REDIRECT' },
+				claims: { sub: 'end' },
+			},
```
[the remark this answers is not in the document]
Comment (noce-nick) [reply packages/cellix/server-oauth2-mock-seedwork/tests/portal-discovery.test.ts:71]: Good suggestion, done

**Comment by noce-nick** [comment]
@sourcery-ai review

**Comment by noce-nick** [comment]
@sourcery-ai review

**Comment by noce-nick** [comment]
@sourcery-ai review

File: .github/copilot-instructions.md [hunk 3]
```diff
@@ -67,7 +67,8 @@ pnpm run gen      # Generate code (e.g., GraphQL types)
 ```
 
 **Important**: 
-- Use `pnpm run verify` to ensure code quality before commits
+- Respect the pre-commit hook and ensure all checks pass while committing. The hook includes the `verify` command, which will prevent the commit on failure. Review the output, fix issues, and retry committing.
+- **NEVER** bypass the pre-commit hook by using `--no-verify` unless explicitly instructed by the human developer. This is a critical safeguard to maintain code quality and security standards.
 - *Note*: Be patient when you run the `verify` command; wait at least 6 minutes for all checks to complete before assuming unexpected failure.
 - Use `pnpm run snyk` to run security scans before commits
 cop- If you encounter this error when attempting to push changes `hook git error: Command failed with exit code 1: git commit -m REDACTED`, it indicates that pre-commit hooks have failed due to unmet code quality or security standards. Review the output from the hooks, fix the reported issues, and try committing again. Be aware that this error has nothing to do with firewall rules or network connectivity; it is expected to fail if your changes do not meet the project's standards.
```
[the remark this answers is not in the document]
Comment (noce-nick) [reply .github/copilot-instructions.md]: Good catch, done

File: packages/cellix/server-oauth2-mock-seedwork/tests/portal-discovery.test.ts [hunk 4]
```diff
@@ -358,4 +430,84 @@ describe('discoverPortalConfigs', () => {
...
+			const portals = discoverPortalConfigs(tmp);
+			expect(portals).toHaveLength(1);
+			expect(warnSpy).toHaveBeenCalled();
+			const p = portals[0];
+			expect(p?.clientId).toBe('cid-nc');
+			expect(p?.redirectUri).toBe('https://nc/redirect');
+		} finally {
+			warnSpy.mockRestore();
+		}
+	});
```
[the remark this answers is not in the document]
Comment (noce-nick) [reply packages/cellix/server-oauth2-mock-seedwork/tests/portal-discovery.test.ts:572]: Outdated, refactored portal discovery to cellix framework package for mock oauth2 server seedwork

File: packages/cellix/server-oauth2-mock-seedwork/src/portal-discovery.ts [hunk 5]
```diff
@@ -35,22 +35,45 @@ export interface PortalOidcConfig {
...
+		const nonConformingWarnState = { fired: false };
+		for (const config of configs) {
+			// Compute registration name: strip ui- prefix from directory and combine with config name
+			const portalDirNameWithoutUiPrefix = name.replace(/^ui-/, '');
+			const registrationName = `${portalDirNameWithoutUiPrefix}-${config.name}`;
+
+			if (!SAFE_NAME_RE.test(registrationName)) {
+				console.warn(
+					`[server-oauth2-mock] Skipping config element "${config.name}" in ${path.join(appsDir, name, 'mock-oidc.json')} — invalid computed registration name "${registrationName}" (must contain only letters, digits, '_' and '-'). Other elements in the same file will still be processed.`,
+				);
```
[the remark this answers is not in the document]
Comment (noce-nick) [reply packages/cellix/server-oauth2-mock-seedwork/src/portal-discovery.ts]: Outdated, refactored portal discovery to cellix framework package for mock oauth2 server seedwork

File: .github/copilot-instructions.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (noce-nick) [reply .github/copilot-instructions.md]: Already fixed

**Comment by noce-nick** [comment]
@sourcery-ai review

File: packages/cellix/server-oauth2-mock-seedwork/src/portal-discovery.ts [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+		const { name } = entry;
+
+		const configs = loadMockOidcConfigs(appsDir, name);
+		if (!configs || configs.length === 0) continue;
+
+		const appDir = path.join(appsDir, name);
+		const parsedEnv = loadAppEnv(appDir, name);
+		if (!parsedEnv) continue;
+
+		const nonConformingWarnState = { fired: false };
```
[the remark this answers is not in the document]
Comment (noce-nick) [reply packages/cellix/server-oauth2-mock-seedwork/src/portal-discovery.ts]: Good suggestion, done

**Comment by noce-nick** [comment]
@sourcery-ai review

File: packages/cellix/server-oauth2-mock-seedwork/src/portal-discovery.ts [hunk 7]
```diff
@@ -0,0 +1,237 @@
...
+		return undefined;
+	}
+
+	return [parsed];
+}
+
+function loadAppEnv(appDir: string, entryName: string): Record<string, string> | null {
+	const envPath = path.join(appDir, '.env');
+	const envLocalPath = path.join(appDir, '.env.local');
+	if (!fs.existsSync(envPath) && !fs.existsSync(envLocalPath)) {
```
[the remark this answers is not in the document]
Comment (noce-nick) [reply packages/cellix/server-oauth2-mock-seedwork/src/portal-discovery.ts]: Good suggestion, done

**Comment by noce-nick** [comment]
@sourcery-ai review

File: packages/cellix/server-oauth2-mock-seedwork/src/portal-discovery.ts [hunk 8]
```diff
@@ -0,0 +1,261 @@
...
+			}
+
+			const portal = buildPortalFromConfig(config, parsedEnv, name, nonConformingWarnState, resolvedEnv);
+			if (!portal) continue;
+
+			// Replace portal.name with computed registrationName
+			const finalPortal: PortalOidcConfig = { ...portal, name: registrationName };
+			portals.push(finalPortal);
+			seenNames.add(registrationName);
+		}
```
[the remark this answers is not in the document]
Comment (noce-nick) [reply packages/cellix/server-oauth2-mock-seedwork/src/portal-discovery.ts:147]: Done

**Comment by noce-nick** [comment]
@sourcery-ai review

**Comment by noce-nick** [comment]
@sourcery-ai review

**Comment by noce-nick** [comment]
@sourcery-ai review

File: packages/cellix/server-oauth2-mock-seedwork/src/utils.ts [hunk 9]
```diff
@@ -76,3 +76,40 @@ export const normalizeOrigin = (value: string) => {
...
+ * // => 'https://mock-auth.example.localhost:8080' (port already set — unchanged)
+ *
+ * ensurePortInUrl('https://secure.example.localhost', 443);
+ * // => 'https://secure.example.localhost' (443 is the HTTPS default — unchanged)
+ * ```
+ */
+export function ensurePortInUrl(baseUrl: string, port: number): string {
+	try {
+		const parsed = new URL(baseUrl);
+		if (!parsed.port && port !== 443) {
```
[the remark this answers is not in the document]
Comment (noce-nick) [reply packages/cellix/server-oauth2-mock-seedwork/src/utils.ts]: Good suggestion, done

**Comment by noce-nick** [comment]
@sourcery-ai review

**Comment by noce-nick** [comment]
@sourcery-ai review

File: azure-pipelines.yml [hunk 10]
```diff
@@ -47,14 +47,14 @@ stages:
...
-        VITE_APP_UI_STAFF_AAD_REDIRECT_URI: $(OCM_APP_UI_STAFF_AAD_REDIRECT_URI_DEV)
-        VITE_APP_UI_STAFF_AAD_SCOPES: $(OCM_APP_UI_STAFF_AAD_SCOPES_DEV)
+        VITE_APP_UI_COMMUNITY_END_USER_B2C_AUTHORITY: $(OCM_APP_UI_COMMUNITY_B2C_AUTHORITY_DEV)
+        VITE_APP_UI_COMMUNITY_END_USER_B2C_CLIENTID: $(OCM_APP_UI_COMMUNITY_B2C_CLIENTID_DEV)
+        VITE_APP_UI_COMMUNITY_END_USER_B2C_REDIRECT_URI: $(OCM_APP_UI_COMMUNITY_B2C_REDIRECT_URI_DEV)
+        VITE_APP_UI_COMMUNITY_END_USER_B2C_SCOPES: $(OCM_APP_UI_COMMUNITY_B2C_SCOPES_DEV)
+        VITE_APP_UI_STAFF_STAFF_USER_AAD_AUTHORITY: $(OCM_APP_UI_STAFF_AAD_AUTHORITY_DEV)
+        VITE_APP_UI_STAFF_STAFF_USER_AAD_CLIENTID: $(OCM_APP_UI_STAFF_AAD_CLIENTID_DEV)
+        VITE_APP_UI_STAFF_STAFF_USER_AAD_REDIRECT_URI: $(OCM_APP_UI_STAFF_AAD_REDIRECT_URI_DEV)
+        VITE_APP_UI_STAFF_STAFF_USER_AAD_SCOPES: $(OCM_APP_UI_STAFF_AAD_SCOPES_DEV)
```
[the remark this answers is not in the document]
Comment (noce-nick) [reply azure-pipelines.yml:57]: This is expected. Portal name is ui-staff and config name is staff-user.

**Comment by noce-nick** [comment]
@sourcery-ai review

**Review by gidich — APPROVED** [review]
